src/core/meshcop/leader.cpp

```diff
@@ -78,7 +79,7 @@ void Leader::HandlePetition(Coap::Header &aHeader, Message &aMessage, const Ip6:
     data.mBorderAgentLocator.SetBorderAgentLocator(HostSwap16(aMessageInfo.GetPeerAddr().mFields.m16[7]));
 
     data.mCommissionerSessionId.Init();
-    data.mCommissionerSessionId.SetCommissionerSessionId(++mSessionId);
+    data.mCommissionerSessionId.SetCommissionerSessionId(mSessionId++);
```

Comment:
`mSessionId` is also used in `Leader::SendPetitionResponse()`.  Post increment here will cause the Commissioner Session IDs in the Network Data and the Petition Response message to be off by one.